test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,356 @@
+	}
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &v1.JaegerList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+	}))
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &esv1.ElasticsearchList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Elasticsearch",
+			APIVersion: "logging.openshift.io/v1",
+		},
+	}))
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+	addToFrameworkSchemeForSmokeTests(t)
+
+}
+
+func (suite *TokenPropagationTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+func (suite *TokenPropagationTestSuite) TestTokenPropagationTest() {
+	// Test parameters
+	const name = "token-prop"
+	const username = "user-test-token"
+	const password = "any"
+	queryName := fmt.Sprintf("%s-query", name)
+	collectorName := fmt.Sprintf("%s-collector", name)
+	collectorPodImageName := "jaeger-collector"
+
+	// create jaeger custom resource
+	exampleJaeger := getESJaegerInstance()
+	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(exampleJaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, collectorName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, queryName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+	route := findRoute(t, fw, name)
+	serviceName := "tokenpropagation"
+
+	host := route.Spec.Host
+	url := fmt.Sprintf("https://%s/api/traces?service=%s", host, serviceName)
+
+	collectorPort := randomPortNumber()
+	collectorPorts := []string{collectorPort + ":14268"}
+	portForwColl, closeChanColl := CreatePortForward(namespace, collectorName, collectorPodImageName, collectorPorts, fw.KubeConfig)
+	defer portForwColl.Close()
+	defer close(closeChanColl)
+	collectorEndpoint := fmt.Sprintf("http://localhost:%s/api/traces", collectorPort)
+
+	cfg := config.Configuration{
+		Reporter:    &config.ReporterConfig{CollectorEndpoint: collectorEndpoint},
+		Sampler:     &config.SamplerConfig{Type: "const", Param: 1},
+		ServiceName: serviceName,
+	}
+	tracer, closer, err := cfg.NewTracer()
+	require.NoError(t, err, "Failed to create tracer in token propagation test")
+
+	tStr := time.Now().Format(time.RFC3339Nano)
+	tracer.StartSpan("TokenTest").
+		SetTag("time-RFC3339Nano", tStr).
+		Finish()
+	closer.Close()
+
+	client, err := oAuthAuthorization(host, username, password)
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
This test is showing a successful response with token propagation. Can we also have a test that is expected to fail - i.e. propagates an invalid token?

Reply:
Yes sure, in that way we can do the test with an invalid token or without the token at all. and we should get a forbidden error.

Reply:
Those tests seem to be missing.